### ISSUE TITLE
Add "From Version" into OED Spec

### ISSUE THREAD

**Issue by benhayes21** [OasisLMF/ODS_OpenExposureData#180]
## Description
Suggestion to add "From Version" column into tabs in the OED spec. 

## Reasons for change
Intention here is to make it easy to see when new fields/values have been added to the spec and from what version they are valid

## Scope of change 
- [ X] Specification
- [ ] Location File
- [ ] Accounts File 
- [ ] Reinsurance Scope 
- [ ] Reinsurance Info 

Propose add column to tabs:
 - OED Input Fields
 - Peril Values
 - Financial Code Values
 - Occupancy Values
 - Construction Values
 - Currency Values
 - Country Values
 - Area Code Values
 - Other Values
 - OED CR Field Appendix

## Impact of change
It shouldn't break anything, but will need to be tested with validation software that uses the specification directly.

**Comment by aiste-kalinauskaite** [comment]
This is a good idea and will be easier to track changes (at least new changes). 

Some additional questions / considerations:

- If the field is changed in any way, would "From version" change as well? 
- Or is this just about implementing the field of when this was first introduced?
- Will the field be populated retrospectively on all that has been released thus far? 
- What version is being tracked here: OED version? when the field is recognised in the financial model (is this ktools?)

**Comment by johcarter** [comment]
Suggest deferring this feature to a later release as it has unanswered questions and potentially very resource intensive.

@benhayes21 @aiste-kalinauskaite @MattDonovan82

**Comment by aiste-kalinauskaite** [comment]
This is needed to gain better clarity of when was released and when the support for it started. Currently. This was a step forward in the right direction. We would like to see this in OED v4.0.0.

**Comment by johcarter** [comment]
I think rather than adding version fields to the specification tables themselves, this would be better implemented as a reporting tool which can read the github history and generate a report of the requested version change information, and probably fits within ODS tools.   The requirements of the tool would need to specified in a bit more detail before it can be considered for implementation.

**Comment by MattDonovan82** [comment]
I agree a reporting tool would be more efficient. Hasn't @sstruzik created a tool that does this?

**Comment by johcarter** [comment]
He has made a start but it needs to be user friendly and accessible, which should be considered as part of the scoping.

**Comment by aiste-kalinauskaite** [comment]
As discussed on today's call - suggestions to have a separate tab to track all the changes for each version and keep appending to the list would be most helpful and would add additional value when reviewing what's changed between the versions.
However, having "From version" column that indicates when each field (or a particular value for the pre-defined lists) was introduced is also important. The data there for a particular column should remain static. E.g. PortNumber was introduced with the first release, so it will have 1.0.0 in it. This data for each column will not change with each release. The only need for change would be if something got assigned in error of when it was introduced. "From version" column would only have a change if a new field is added and it would get assigned a version number of when it is done. Afterwards, no further change to that column is expected.
We discussed that once you have a log of all the changes, perhaps it could be automated from there to populate "From Version" column for the first time.
We would like to see this in OED v4.0.0 release.

**Comment by aiste-kalinauskaite** [comment]
On a separate note (although this should be a different ticket), it would also be very useful, I am just not sure where it should sit:

information when the field is recognised in the financial model - i.e. the version of the ktools and of worker.
